LayoutTests/TestExpectations

```diff
@@ -6745,3 +6745,6 @@ webkit.org/b/245032 imported/w3c/web-platform-tests/css/compositing/root-element
 webkit.org/b/261916 media/video-pause-while-seeking.html [ Pass Failure ]
 
 webkit.org/b/262734 http/tests/security/service-worker-network-load.html [ Pass Timeout ]
+
+# Pointer Lock unadjustedMovement option is not implemented on all platforms.
```

Comment:
This should be represented using platform-specific expectations in `LayoutTests/platform/imported...`.

Reply:
Sorry, I am not a regular contributor to WebKit and I need a little hand-holding in this area. I am not sure what you're asking for here.

What I'm trying to represent is that this test passes only on Mac.

Should I ...

1. Change this line in LayoutTests/TestExpections to be just [ Failure ] and then the LayoutTests/platform/mac/TestExpectations contains an entry with just [ Pass ]? In other words, is your objection that [ Pass Failure ] in the top level TestExpectations representing that this test is flaky but really it's just platform implementation differences?
2. Remove this line from LayoutTests/TestExpectations and instead add [ Failure ] entries for this test in every LayoutTests/platform/*/TestExpections file (except LayoutTests/platform/mac/TestExpectations) where it is [ Pass ]?
3. Remove all changes to TestExpectations files and instead check in LayoutTests/platform/*/imported/w3c/web-platform-tests/pointerlock/pointerlock_unadjustedMovement-expected.txt where the contents of that file show the failure on unsupported platforms?
4. Something else?

Reply:
No worries! I meant (3), since that will make it easier to detect progressions and better document behavior. You are right that `[ Pass Failure ]` represents a flaky test, and is not ideal.

Reply:
You can leave the Mac expectation next to the test though. Just add different expectations for the failing platforms.

Reply:
I pushed an update that attempts to accomplish this. I may need to add some more pointerlock_unadjustedMovement-expected.txts to various platforms. We will see what EWS thinks about it.

One thing I ran into with actually running this test on all platforms that support pointerlock rather than skipping it on platforms that don't support unadjustedMovement is that the test (LayoutTests/imported/w3c/web-platform-tests/pointerlock/pointerlock_unadjustedMovement.html) as I originally wrote it would lead to timeouts. This is because the wpt test harness doesn't handle assertion failures / exceptions thrown from an async step_func well; it just ignores them and then the test times out. Given that I didn't want the tests to timeout and slow everything down, I had to modify the test to use a non-async step_func and use then/catch on the promise returned from requestPointerLock.